crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -533,6 +535,18 @@ fn check_visibility_and_signature<E: fmt::Debug, S: StorageView<E>>(
             .load_type(ty)
             .map_err(|e| convert_type_argument_error(idx, e, context.vm, context.state_view))?;
     }
+    if from_init {
```

Comment:
cc @dariorussi @amnn as this might be important to consider when thinking about package upgrades